File: Python/sorts/selection.py
arr = [7, 8, 9, 4, 6, 1, 0]


# Функция для поиска наименьшего числа в массиве
def findSmallest(arr):
    smallest = arr[0]
    smallest_index = 0

    for i in range(1, len(arr)):
        if smallest > arr[i]:
            smallest = arr[i]
            smallest_index = i
    return smallest_index


# Сортировка выборкой
def selectionSort(arr):
    new_arr = []
    for i in range(len(arr)):
        smallest = findSmallest(arr)
        new_arr.append(arr.pop(smallest))
        print(arr, " - ", new_arr)
    return new_arr


print("Сортировка выбором:")
print("Исходный массив:", arr)
print("Отсортированный массив:", selectionSort(arr))

File: Java/sorts/selection.java
import java.util.ArrayList;

public class selection{
    
    public static int findSmallest(ArrayList<Integer> array){
        int smallest = array.get(0);
        int smallest_index = 0;

        for(int i = 1; i < array.size(); i++){
            if (smallest < array.get(i)) {
                smallest = array.get(i);
                smallest_index = i;
            }
        }
        return smallest_index;
    }

    public static ArrayList<Integer> selectionSort(ArrayList<Integer> array){
        ArrayList<Integer> new_arr = new ArrayList<>();
        int size = array.size();       // Если array.size() вставить в цикл, то цикл будет брать длину каждую иттерацию

        for(int i = 0; i < size; i++){
            int smallest = findSmallest(array);
            new_arr.add(array.remove(smallest));
        }
        return new_arr;
    }

    public static void main(String[] args){

        ArrayList<Integer> arr = new ArrayList<>();
        arr.add(7);
        arr.add(8);
        arr.add(9);
        arr.add(4);
        arr.add(6);
        arr.add(1);
        arr.add(0);

        System.out.println(arr);
        System.out.println(selectionSort(arr));

    }
}